src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -71,6 +72,26 @@ class PrisonerIepLevelReviewService(
     return addIepLevel(prisonerInfo, iepReview)
   }
 
+  @Transactional
+  suspend fun addIepMigration(bookingId: Long, iepMigration: IepMigration): IepDetail {
+    val prisonerInfo = prisonApiService.getPrisonerInfo(bookingId)
+    return prisonerIepLevelRepository.save(
+      PrisonerIepLevel(
+        iepCode = iepMigration.iepLevel,
+        commentText = iepMigration.comment,
+        bookingId = prisonerInfo.bookingId,
+        prisonId = iepMigration.establishmentCode,
+        locationId = iepMigration.locationId,
+        sequence = 0,
+        current = iepMigration.current,
+        reviewedBy = iepMigration.userId,
+        reviewTime = iepMigration.iepTime,
+        reviewType = iepMigration.reviewType,
+        prisonerNumber = prisonerInfo.offenderNo
+      )
+    ).translate()
```

Comment:
> I'm a bit puzzled by this translate() method call?

It's [this function](https://github.com/ministryofjustice/hmpps-incentives-api/blob/bec6c173e8f49602d01ea8aeaa12423bc5e88573/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt#L231-L247) which is an [extension function](https://kotlinlang.org/docs/extensions.html)

- `dto` - [data transfer object](https://en.wikipedia.org/wiki/Data_transfer_object)
- `jpa` - [jakarta persistence API](https://www.infoworld.com/article/3379043/what-is-jpa-introduction-to-the-java-persistence-api.html)

We translate it because we don't want to expose `jpa` fields (`PrisonerIepLevel`) details to clients, instead we map it to `IepDetail` (which is part of the `dto` package)

>  is this value actually sent back to the client?

Yes it should be as `migrateIepReview` returns an `IepDetail`

Hope that makes sense 😄

Reply:
Thanks, that's very clear ☺️ 

Oh I see, extension functions is basically monkey-patching 👌 - it's adding a new method to an existing class (`PrisonerIepLevel` in this case)

And it makes sense to keep the internal DB representation (`jpa.*`) private and separate from the public API representation (`dto.*`) 👌